internal/clientconn/conninfo/conn_info.go

```diff
@@ -35,12 +37,14 @@ type ConnInfo struct {
 	password     string // protected by rw
 	metadataRecv bool   // protected by rw
 
+	sc *scram.ServerConversation // protected by rw
+
 	// If true, backend implementations should not perform authentication
 	// by adding username and password to the connection string.
 	// It is set to true for background connections (such us capped collections cleanup)
 	// and by the new authentication mechanism.
 	// See where it is used for more details.
-	BypassBackendAuth bool
+	bypassBackendAuth bool
```

Comment:
```suggestion
	bypassBackendAuth bool // protected by rw
```

Normally, there would be a [mutex hat](https://dmitri.shuralyov.com/idiomatic-go/entries/2), but to make this struct smaller the order of fields is weird. So we should have a comment